packages/auth/src/error/types/validation.ts

```diff
@@ -0,0 +1,66 @@
+
+export enum AuthValidationErrorCode {
+	EmptySignInUsername = 'EmptySignInUsername',
+	EmptySignInPassword = 'EmptySignInPassword',
+	EmptySignUpUsername = 'EmptySignUpUsername',
+	EmptySignUpPassword = 'EmptySignUpPassword',
+	EmptyConfirmSignUpUsername = 'EmptyConfirmSignUpUsername',
+	EmptyConfirmSignUpCode = 'EmptyConfirmSignUpCode',
+	EmptyResendSignUpCodeUsername = 'EmptyresendSignUpCodeUsername',
+	EmptyChallengeResponse = 'EmptyChallengeResponse',
+	EmptyConfirmResetPasswordUsername = 'EmptyConfirmResetPasswordUsername',
+	EmptyConfirmResetPasswordNewPassword = 'EmptyConfirmResetPasswordNewPassword',
+	EmptyConfirmResetPasswordConfirmationCode = 'EmptyConfirmResetPasswordConfirmationCode',
+	EmptyResetPasswordUsername = 'EmptyResetPasswordUsername',
+}
+
+export type AuthValidationErrorMap = {
+	[AuthValidationErrorCode.EmptyChallengeResponse]: {
+		message: 'challengeResponse is required to confirmSignIn';
+		recovery: 'Make sure that a valid challenge response is passed for confirmSignIn';
+	};
+	[AuthValidationErrorCode.EmptyConfirmResetPasswordUsername]: {
+		message: 'username is required to confirmResetPassword';
+		recovery: 'Make sure that a valid username is passed for confirmResetPassword';
+	};
+	[AuthValidationErrorCode.EmptyConfirmSignUpCode]: {
+		message: 'code is required to confirmSignUp';
+		recovery: 'Make sure that a valid code is passed for confirmSignUp';
+	};
+	[AuthValidationErrorCode.EmptyConfirmSignUpUsername]: {
+		message: 'Username is required to confirmSignUp';
```

Comment:
```suggestion
		message: 'username is required to confirmSignUp';
```

Just to make it similar to all the other ones (same case), few more below